src/torchcodec/_core/FFMPEGCommon.cpp

```diff
@@ -88,21 +88,67 @@ void setDefaultChannelLayout(
 #endif
 }
 
-void setChannelLayout(
-    UniqueAVFrame& dstAVFrame,
-    const UniqueAVCodecContext& avCodecContext) {
+void setDefaultChannelLayout(UniqueAVFrame& avFrame, int numChannels) {
 #if LIBAVFILTER_VERSION_MAJOR > 7 // FFmpeg > 4
-  auto status = av_channel_layout_copy(
-      &dstAVFrame->ch_layout, &avCodecContext->ch_layout);
-  TORCH_CHECK(
-      status == AVSUCCESS,
-      "Couldn't copy channel layout to avFrame: ",
-      getFFMPEGErrorStringFromErrorCode(status));
+  AVChannelLayout channel_layout;
+  av_channel_layout_default(&channel_layout, numChannels);
+  avFrame->ch_layout = channel_layout;
 #else
-  dstAVFrame->channel_layout = avCodecContext->channel_layout;
-  dstAVFrame->channels = avCodecContext->channels;
+  uint64_t channel_layout = av_get_default_channel_layout(numChannels);
+  avFrame->channel_layout = channel_layout;
+  avFrame->channels = numChannels;
+#endif
+}
 
+void validateNumChannels(const AVCodec& avCodec, int numChannels) {
+#if LIBAVFILTER_VERSION_MAJOR > 7 // FFmpeg > 4
+  if (avCodec.ch_layouts == nullptr) {
+    // If we can't validate, we must assume it'll be fine. If not, FFmpeg will
+    // eventually raise.
+    return;
+  }
+  for (auto i = 0; avCodec.ch_layouts[i].order != AV_CHANNEL_ORDER_UNSPEC;
```

Comment:
I'm not sure that this is correct. I think it will often work, but [the docs say](https://ffmpeg.org/doxygen/6.1/structAVCodec.html#a710e3bd3081124ef3364b0c520379dd8) that `ch_layouts` is:
> Array of supported channel layouts, terminated with a zeroed layout.

That does imply that the terminating struct will have `AV_CHANNEL_ORDER_UNSPEC` for its order because that corresponds to 0 in the `AVChannelOrder` enum, but I think that's also a valid order for a real layout.

I think it may be safer to say `avCodec.ch_layouts[i] != AVChannelLayout{0}`. That's also not obvious, so we could define `const auto emptyAVChannelLayout = AVChannelLayout{0}` and use that in the comparison.